### PR TITLE
Fix/copy token

### PR DESCRIPTION
**Pull request by alexvanin**
Return copy of the session token from cache for safe re-signing operations.

### CODE REVIEW

File: pool/pool.go [hunk 1]
```diff
@@ -1582,3 +1582,30 @@ func newAddressFromCnrID(cnrID *cid.ID) *address.Address {
 	addr.SetContainerID(cnrID)
 	return addr
 }
+
+func copySessionTokenWithoutSignature(from session.Token) (to session.Token) {
```
Comment (KirillovDenis) [inline pool/pool.go:1586]: Actually, the context is also not copied

**Review by KirillovDenis — APPROVED** [review]

File: pool/pool.go (hunk 1 above)
Comment (alexvanin) [reply pool/pool.go]: Updated function name and test.

**Comment by alexvanin** [comment]
> Return from cache by value is simpler, isn't it? I guess it'd cause some refactoring, so it can be done later.

The issue with the signature slice that modifies `token.Sign()` function.

**Comment by alexvanin** [comment]
> > Return from cache by value is simpler, isn't it? I guess it'd cause some refactoring, so it can be done later.
> 
> The issue with the signature slice that modifies `token.Sign()` function.

However it might work. We had an issues with copies in `tokToSign` copy. Maybe fix on a cache level might solve it? Let me check.

**Comment by alexvanin** [comment]
> > > Return from cache by value is simpler, isn't it? I guess it'd cause some refactoring, so it can be done later.
> > 
> > 
> > The issue with the signature slice that modifies `token.Sign()` function.
> 
> However it might work. We had an issues with copies in `tokToSign` copy. Maybe fix on a cache level might solve it? Let me check.

I was wrong, no the the signature but session context is the problem there. I did update the tests.

**Pull request merged**